File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    let answer = [numer1 * denom2 + numer2 * denom1, denom1 * denom2];
    return answer.map((el) => el / getGCD(...answer));
}

const getGCD = (num1, num2) => {
    if (num1 % num2 === 0) return num2; 
    
    const remainder = num1 % num2;
    return getGCD(num2, remainder);
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.kt
class Solution {
    fun solution(numer1: Int, denom1: Int, numer2: Int, denom2: Int): IntArray {
        val denom = denom1 * denom2
        val numerSum = numer1 * denom2 + numer2 * denom1
        
        val gcdOfNumerAndDenom = getGCD(denom, numerSum)
        println(gcdOfNumerAndDenom)
        
        return intArrayOf(numerSum / gcdOfNumerAndDenom, denom / gcdOfNumerAndDenom)
    }
    
    fun getGCD(a: Int, b: Int): Int {
    var num1 = a
    var num2 = b

    while (num2 != 0) {
        val (newNum1, newNum2) = Pair(num2, num1 % num2)
        num1 = newNum1
        num2 = newNum2
    }
    return num1
}
}